zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/CamundaOpenApiController.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.nio.charset.StandardCharsets;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@CamundaRestController
+public class CamundaOpenApiController {
+
+  @GetMapping(value = "/camunda-api-docs", produces = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<String> loadCamundaApiDocs() {
+    try {
+      final ClassPathResource resource = new ClassPathResource("apidoc/rest-api.yaml");
+
+      final String yamlContent =
+          new String(resource.getInputStream().readAllBytes(), StandardCharsets.UTF_8);
+
+      final ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+      final Object obj = yamlReader.readValue(yamlContent, Object.class);
+
+      final ObjectMapper jsonWriter = new ObjectMapper();
+      final String jsonContent = jsonWriter.writeValueAsString(obj);
```

Comment:
@danielkelemen, since you have worked on the object mapper lately, don't we have an instance we can simply reuse here? I can hardly believe that creating new object mappers here again is a good idea 🙂

Reply:
You could just inject the default OM but that comes without the `YAMLFactory`, I guess that's important.

Maybe that could be added to the `gatewayRestObjectMapper`? Or could that be a problem for other places where we use the OM? I'm not sure.

Reply:
Ah, there is two! I haven't noticed the second OM 😄
For the `YAMLFactory` you could define a new OM bean so it's not recreated every time the Controller is called.